File: r/triangle/_iterations/1/triangle.R
triangle <- function(x, y, z) {
  sides <- sort(c(x, y, z))
  stopifnot(all(sides > 0) & sides[1] + sides[2] >= sides[3])
  kind <- 42
  if (sides[1] == sides[2] & sides[2] == sides[3]) {
    class(kind) <- c(class(kind), "equilateral")
  }
  if (sides[1] == sides[2] | sides[2] == sides[3]) {
    class(kind) <- c(class(kind), "isosceles")
  }
  if (sides[1] != sides[2] & sides[2] != sides[3]) {
    class(kind) <- c(class(kind), "scalene")
  }
  kind
}

File: javascript/triangle/_iterations/1/triangle.js
export class Triangle {
    constructor(side1, side2, side3) {
        this.sides = [side1, side2, side3].sort((a, b) => a - b);
    }

    kind() {
        if (!this.sides.every(n => n > 0) || (this.sides[0] + this.sides[1]) < this.sides[2]) {
            throw new Error('Triangle violates triangle inequality');
        }
        if (this.sides[0] == this.sides[1] && this.sides[1] == this.sides[2]) {
            return 'equilateral';
        } else if (this.sides[0] == this.sides[1] || this.sides[1] == this.sides[2]) {
            return 'isosceles';
        } else {
            return 'scalene';
        }
    }
}
